apps/hvn/src/components/donation/donation-button.tsx

```diff
@@ -0,0 +1,50 @@
+'use client'
+import { FC, useEffect } from 'react'
+
+import { Button, ButtonProps } from '@fc/ui/base/button'
+
+import { usePathname, useSearchParams } from 'next/navigation'
+import { toast } from 'sonner'
+
+type DonateButtonProps = {
+  amount: number
+  email: string
+  type: 'monthly' | 'one-time'
+} & Omit<ButtonProps, 'type' | 'email'>
+
+const DonateButton: FC<DonateButtonProps> = ({ amount, children, email, type, ...props }) => {
+  const path = usePathname()
+  const searchParams = useSearchParams()
+  const status = searchParams.get('status')
+  useEffect(() => {
+    if (status === 'success') {
+      toast.success('Payment Successful', { description: 'Thank you for your donation!' })
+    } else if (status === 'cancel') {
+      toast.error('Payment Canceled', { description: 'Your donation was not completed.' })
+    }
+  }, [status])
+
+  const handleDonate = async () => {
+    const successUrl = `${window.location.origin}${path}?status=success`
+    const cancelUrl = `${window.location.origin}${path}?status=cancel`
```

Comment:
Would it be better to create a separate page for payment status? In this case we can move the status logic from homepage.